GameOn-Frontend/app/(tabs)/profile/index.tsx

```diff
@@ -27,9 +27,25 @@ export default function Profile() {
   const router = useRouter();
   const { signOut } = useAuth();
 
-  const { isLoaded, isSignedIn, user } = useUser();
+  const { isLoaded, isSignedIn, user } = useUser(); 
+  
+  if ((!isLoaded || !isSignedIn) && !__DEV__) return null;
 
-  if (!isLoaded || !isSignedIn) return null;
+const useMockUser = __DEV__ && !isSignedIn;
+
+const activeUser = useMockUser
+  ? {
+      fullName: userDetails.name,
+      primaryEmailAddress: { emailAddress: userDetails.email },
+      imageUrl: userDetails.image,
+    }
+  : {
+      fullName: user?.fullName ?? "",
+      primaryEmailAddress: {
+        emailAddress: user?.primaryEmailAddress?.emailAddress ?? "",
+      },
+      imageUrl: user?.imageUrl ?? images.defaultProfile,
+    };
```

Comment:
It looks like we can simplify the user object by keeping a consistent shape for email and removing the nested primaryEmailAddress field. This keeps the mock and the real user data aligned and reduces unnecessary nesting.

```suggestion
const activeUser = useMockUser
  ? {
      fullName: userDetails.name,
      emailAddress: userDetails.email,
      imageUrl: userDetails.image,
    }
  : {
      fullName: user?.fullName ?? "",
      emailAddress: user?.primaryEmailAddress?.emailAddress ?? "",
      imageUrl: user?.imageUrl ?? images.defaultProfile,
    };
```